org.eclipse.scout.rt.opentelemetry.sdk/src/main/java/org/eclipse/scout/rt/opentelemetry/sdk/traces/RuleBasedSampler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.opentelemetry.sdk.traces;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import org.eclipse.scout.rt.opentelemetry.sdk.traces.RuleBasedSamplerBuilder.ISamplingCondition;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+/***
+ * Makes a sample decision based on the {@link ScoutSamplingRule}
+ *
+ * Inspired by <a href="https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/samplers/src/main/java/io/opentelemetry/contrib/sampler/RuleBasedRoutingSampler.java">RuleBasedRoutingSampler</a>
+ */
+public class RuleBasedSampler implements Sampler {
```

Comment:
Should we define this class as `final` - otherwise we should consider changing the ctor to `protected` if this class can be extended. Also there are no getters/setters for the member attributes (which would not be a problem if it is `final`).